app/rdf/query-search.ts

```diff
@@ -238,11 +237,16 @@ export const searchCubes = async ({
       )`
           : ""
       }
-      
   `.GROUP().BY`?iri`.THEN.BY`?title`.THEN.BY`?status`.THEN.BY`?datePublished`
     .THEN.BY`?description`.THEN.BY`?publisher`.THEN.BY`?creatorIri`.THEN
     .BY`?creatorLabel`.prologue`${pragmas}`;
 
+  if (themeValues.length) {
+    scoresQuery = scoresQuery.HAVING`${themeValues
```

Comment:
@bprusinowski what was the issue with `FILTER()` ? - can we find a fix for that one?

Reply:
But it doesn't look like it affected the performance in any noticeable way

Reply:
OK, you don't want to loose the other themes on the selected cubes, right?
Instead of bad performing `HAVING` you may state the theme i.e. `?iri dcat:theme <https://register.ld.admin.ch/opendataswiss/category/agriculture> .`
this will bring the desired filtering effect, not?

Reply:
Yup, you are right – I updated the query 🎉